### PR TITLE
Krill freq-differencing, hake masking, and glider AZFP notebooks

### PR DESCRIPTION
**Pull request by ctuguinay**
Addresses #51

**TODO(s)**:

- [x] Krill NASC Notebook using Frequency Differencing
    - Perhaps I make this notebook even bigger? Just as an example of what Echopype and Dask are capable of in terms of mass processing.
        - This will differentiate it from the Cruise Tracks notebook
        - Decided to use 4 days instead of 2 days like what's in the Cruise Tracks notebook
- [x] Hake NASC Notebook using Echoregions
    - Does this notebook need Dask?
        - I think I'm going to keep it here even though it's probably not needed
        - Even with Dask, the Krill notebook may not be able to run since it's so large, but this example will (almost) always work since it's just a few files
- [x] Revamp any of the older notebooks so that they work with the latest version of Echopype
    - Didn't do much revamping since I thought, for the most part, they are good as is.
    - The Hake Survey Cruisetracks notebook is slow
        - It would be faster to do the following: `Parse Echodata objects -> Combine Echodata objects -> Compute Sv -> Compute MVBS` instead of `Parse Echodata objects -> Compute Sv for all Echodata Objects -> Compute MVBS for all Sv datasets -> Combine MVBS Datasets` but I think that there is benefit in showing that the latter is possible. Especially since that may be the sort of (psuedo-)realtime streaming situation that one may want to emulate while on a ship.
- [x] Add Krill and Hake Notebook summaries to `index.md`
- [x] List which notebooks can run on the limited computing resources of Binder
    - Used `Binder Friendly` and `Not Binder Friendly`
        - Only Echopype Tour and Hake NASC are binder friendly since they use less than a GiB of RAW data.
- [x] Add the Glider AZFP notebook based on PR #31 by Emilio. Incorporate the changes made in https://github.com/OSOceanAcoustics/echopype/pull/1369 to plot depth accurately in the Echogram.
    - Also added shoal masking example from Echopy

TODOs from Wu-Jung's PR:
- krill NASC: 
  - [x] review the Phillips et al paper re how NASC integration is done there, to make sure the ops are in good shape
- hake NASC:
  - [x] change the sequence to: masked Sv -> Regrid masked Sv -> Compute NASC, so that the bottom contamination is removed
  - [x] add a 1 m bottom offset to demonstrate flexibility (and that is how FEAT uses the bottom annotation too)
- cruise track:
  - [x] add depth to the Sv datasets
  - [x] use ping_time intervals for MVBS processing (I am actually not sure why the current processing wouldn't fail, maybe it's just pure luck that no MVBS timestamp overlap occurs in this segment? those that's a lot of files to have this luck)
  - [x] change the vertical axis to be "depth" in the final MVBS plots
- [x] I think we should remove the `echopype_tour` notebook? the coverage of function is not comprehensive and all the ingredients are already in other notebooks
    - I agree. I removed this

Another pass after the above changes to pick up small loose ends:
- [x] consistency across all notebooks on how sections are defined
- [x] emphasis and explanation on out-of-core computation

### CODE REVIEW

**Comment by leewujung** [comment]
small notes when i happen to see what commit flew by - I don't think echopy is on conda? I haven't checked lately, but if that's the case it has to get pip installed.

**Comment by ctuguinay** [comment]
@leewujung It should be here: https://anaconda.org/conda-forge/echopy. On my end, I could also run `conda install echopy`.

File: notebooks/krill_NASC.ipynb [hunk 1]
```diff
@@ -0,0 +1,803 @@
+{
```
Comment (valentina-s) [inline notebooks/krill_NASC.ipynb:1]: <p>simplify to "out-of-core and parallelized" to distributed computation and remove lazy-load? The idea is to remove some of the jargon in the beginning of the notebook (if people pick this notebook as the first one to read it may be hard to follow and they get discouraged.)</p><p> </p><p>There are also more details on what lazy load is below.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/OSOceanAcoustics/echopype-examples/pull/52/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (valentina-s) [inline notebooks/krill_NASC.ipynb:1]: <p>Could link to lazy loading reference: <a href="https://en.wikipedia.org/wiki/Lazy_loading" target="_blank">https://en.wikipedia.org/wiki/Lazy_loading</a>. If people first open this notebook, they may not know what these terms mean.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/OSOceanAcoustics/echopype-examples/pull/52/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (valentina-s) [inline notebooks/krill_NASC.ipynb:1]: <p>Can add how long it took on your machine so users have some idea of how long it can take (especially if they have reset it and do not see the outputs of your run)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/OSOceanAcoustics/echopype-examples/pull/52/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Comment (valentina-s) [inline notebooks/krill_NASC.ipynb:1]: <p>on a map</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/OSOceanAcoustics/echopype-examples/pull/52/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='34'/>

File: notebooks/ship_track.ipynb [hunk 2]
```diff
@@ -0,0 +1,2500 @@
+{
```
Comment (valentina-s) [inline notebooks/ship_track.ipynb:1]: <p>can make the name of the link to the env file as <a href="https://github.com/OSOceanAcoustics/echopype-examples/blob/main/conda/environment-ep-main.yml" target="_blank">conda/environment-ep-main.yml</a>.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/OSOceanAcoustics/echopype-examples/pull/52/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (valentina-s) [inline notebooks/ship_track.ipynb:1]: <p>power, ease of use, and familiarity (I am not sure what is meant by familiarity?)</p><p> </p><p>Two days ... Make echopype as <code>echopype</code> or Echopype and be consistent</p><p> </p><p>Note that there is a deviation from the Sonar Netcdf convention: is there some link in the ediscussion of echodata levels?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/OSOceanAcoustics/echopype-examples/pull/52/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (valentina-s) [inline notebooks/ship_track.ipynb:1]: <p>Add a text cell above to state why this step is needed (and where this metadata comes from)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/OSOceanAcoustics/echopype-examples/pull/52/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (valentina-s) [inline notebooks/ship_track.ipynb:1]: <p>can you state what is the approach here vs the combining of the individual EchoData objects</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/OSOceanAcoustics/echopype-examples/pull/52/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Comment (valentina-s) [inline notebooks/ship_track.ipynb:1]: <p>Explain the upcoming steps so people have an idea before reading the code.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/OSOceanAcoustics/echopype-examples/pull/52/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

Comment (valentina-s) [inline notebooks/ship_track.ipynb:1]: <p>based on</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/OSOceanAcoustics/echopype-examples/pull/52/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='36'/>

Comment (valentina-s) [inline notebooks/ship_track.ipynb:1]: <p>Link OOI with Ocean Observatories Initiative website.  Also link to the OOI_eclipse notebook</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/OSOceanAcoustics/echopype-examples/pull/52/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='42'/>

File: notebooks/OOI_eclipse.ipynb [hunk 3]
```diff
@@ -0,0 +1,11576 @@
+{
```
Comment (valentina-s) [inline notebooks/OOI_eclipse.ipynb:1]: <p>Make link to yml relative</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/OSOceanAcoustics/echopype-examples/pull/52/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (valentina-s) [inline notebooks/OOI_eclipse.ipynb:1]: <p>out-of-core and parallelized -&gt; distributed</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/OSOceanAcoustics/echopype-examples/pull/52/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (valentina-s) [inline notebooks/OOI_eclipse.ipynb:1]: <p>link to lazy loading reference (and some explanation of what it means)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/OSOceanAcoustics/echopype-examples/pull/52/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

Comment (ctuguinay) [reply notebooks/OOI_eclipse.ipynb:1]: <p>Done</p>

Comment (ctuguinay) [reply notebooks/OOI_eclipse.ipynb:1]: <p>Done</p>

Comment (ctuguinay) [reply notebooks/OOI_eclipse.ipynb:1]: <p>Done</p>

**Comment by ctuguinay** [comment]
<p>Done</p>

---

 <a href='https://app.reviewnb.com/OSOceanAcoustics/echopype-examples/pull/52/discussion/#comment-2270066067'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by ctuguinay** [comment]
<p>Done (and for all other notebooks too)</p>

---

 <a href='https://app.reviewnb.com/OSOceanAcoustics/echopype-examples/pull/52/discussion/#comment-2270066081'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by ctuguinay** [comment]
<p>I think I will just use EchoData without the <code> </code></p>

---

 <a href='https://app.reviewnb.com/OSOceanAcoustics/echopype-examples/pull/52/discussion/#comment-2270066101'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by ctuguinay** [comment]
<p>Done</p>

---

 <a href='https://app.reviewnb.com/OSOceanAcoustics/echopype-examples/pull/52/discussion/#comment-2270066090'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by ctuguinay** [comment]
<p>Moved it below</p>

---

 <a href='https://app.reviewnb.com/OSOceanAcoustics/echopype-examples/pull/52/discussion/#comment-2270066113'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by ctuguinay** [comment]
<p>Yeah it is just individual points</p>

---

 <a href='https://app.reviewnb.com/OSOceanAcoustics/echopype-examples/pull/52/discussion/#comment-2270066128'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by ctuguinay** [comment]
<p>Made this change</p>

---

 <a href='https://app.reviewnb.com/OSOceanAcoustics/echopype-examples/pull/52/discussion/#comment-2270066172'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by ctuguinay** [comment]
<p>I agree. I'll use 'Align echosounder data according to depth'</p>

---

 <a href='https://app.reviewnb.com/OSOceanAcoustics/echopype-examples/pull/52/discussion/#comment-2270066185'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by ctuguinay** [comment]
<p>made this change</p>

---

 <a href='https://app.reviewnb.com/OSOceanAcoustics/echopype-examples/pull/52/discussion/#comment-2270066163'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by ctuguinay** [comment]
<p>Done</p>

---

 <a href='https://app.reviewnb.com/OSOceanAcoustics/echopype-examples/pull/52/discussion/#comment-2270066153'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by ctuguinay** [comment]
<p>Done</p>

---

 <a href='https://app.reviewnb.com/OSOceanAcoustics/echopype-examples/pull/52/discussion/#comment-2270066202'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by ctuguinay** [comment]
<p>Done</p>

---

 <a href='https://app.reviewnb.com/OSOceanAcoustics/echopype-examples/pull/52/discussion/#comment-2270066209'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by ctuguinay** [comment]
<p>made this change</p>

---

 <a href='https://app.reviewnb.com/OSOceanAcoustics/echopype-examples/pull/52/discussion/#comment-2270066223'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by ctuguinay** [comment]
<p>Done</p>

---

 <a href='https://app.reviewnb.com/OSOceanAcoustics/echopype-examples/pull/52/discussion/#comment-2270066238'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by ctuguinay** [comment]
<p>Done</p>

---

 <a href='https://app.reviewnb.com/OSOceanAcoustics/echopype-examples/pull/52/discussion/#comment-2270066258'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

File: notebooks/krill_NASC.ipynb (hunk 1 above)
Comment (ctuguinay) [reply notebooks/krill_NASC.ipynb:1]: <p>Added</p>

Comment (ctuguinay) [reply notebooks/krill_NASC.ipynb:1]: <p>Added</p>

Comment (ctuguinay) [reply notebooks/krill_NASC.ipynb:1]: <p>added</p>

Comment (ctuguinay) [reply notebooks/krill_NASC.ipynb:1]: <p>added</p>

File: notebooks/ship_track.ipynb (hunk 2 above)
Comment (ctuguinay) [reply notebooks/ship_track.ipynb]: <p>I made it relative as your other comment suggested</p>

Comment (ctuguinay) [reply notebooks/ship_track.ipynb]: <p>removed the goal section</p>

Comment (ctuguinay) [reply notebooks/ship_track.ipynb]: <p>I'll just use Echopype from now on</p>

Comment (ctuguinay) [reply notebooks/ship_track.ipynb]: <p>also the sonar netcdf part was cut in order to make the description simpler</p>

Comment (ctuguinay) [reply notebooks/ship_track.ipynb]: <p>It actually isn't needed...and I'm not quite sure where it comes from other than the NOAA Fisheries website link</p>

Comment (ctuguinay) [reply notebooks/ship_track.ipynb]: <p>Added a markdown statement for this</p>

Comment (ctuguinay) [reply notebooks/ship_track.ipynb]: <p>added it</p>

Comment (ctuguinay) [reply notebooks/ship_track.ipynb]: <p>added it</p>

Comment (ctuguinay) [reply notebooks/ship_track.ipynb]: <p>added this</p>

Comment (ctuguinay) [reply notebooks/ship_track.ipynb]: <p>added</p>

**Comment by ctuguinay** [comment]
<p>Made this change</p>

---

 <a href='https://app.reviewnb.com/OSOceanAcoustics/echopype-examples/pull/52/discussion/#comment-2270066141'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by ctuguinay** [comment]
@leewujung Just reran with 0.9.0

**Comment by leewujung** [comment]
@ctuguinay : Awesome! Merging now!

**Review by leewujung — APPROVED** [review]
